resource/decker/support/plugins/handout/handout.js

```diff
@@ -0,0 +1,571 @@
+    if (offset < 0) {
+      visible = box.bottom - host.top;
+    } else {
+      visible = host.bottom - box.top;
+    }
+    if (!mostVisible || mostVisibleValue < visible) {
+      mostVisible = slide;
+      mostVisibleValue = visible;
+    }
+  }
+  /* If the current central slide changed */
+  if (centralSlide !== mostVisible) {
+    centralSlide = mostVisible;
+    // Inform feedback plugin (load questions)
+    const feedback = Reveal.getPlugin("feedback");
+    if (feedback && feedback.getEngine().api) {
+      feedback.requestMenuContent(mostVisible);
+    }
+    // Inform menu plugin (highlight current slide)
+    const menu = Reveal.getPlugin("decker-menu");
+    if (menu) {
+      menu.updateCurrentSlideMark(mostVisible);
+    }
+    // Inform decker plugin (index page)
+    const decker = Reveal.getPlugin("decker");
+    if (decker && mostVisible.dataset.hIndex) {
+      decker.updateLastVisitedSlide({ h: Number(mostVisible.dataset.hIndex) });
+      decker.updatePercentage(Number(mostVisible.dataset.hIndex));
+    }
+  }
+}
+
+/**
+ * Create an intersection observer, if it not already exists,
+ * that manages the slides that are currently visible so the
+ * onscroll callback can be optimised.
+ * @param {*} slideElementList
+ * @returns
+ */
+function createVisibleSlideIntersectionObserver(slideElementList) {
+  // Do not reinitialise if we already entered handout mode in the past
+  if (visibleSlideIntersectionObserver) {
+    return;
+  }
+  // Callback to add or remove slide to the list of currently visible slides
+  const visibilityCallback = function (entries, observer) {
+    // Do nothing if we left handout mode
+    if (!handoutSlideMode) return;
+    for (const entry of entries) {
+      if (entry.isIntersecting) {
+        visibleSlides.add(entry.target);
+      } else {
+        visibleSlides.delete(entry.target);
+      }
+    }
+  };
+  // Only trigger if a section becomes partly visible or disappears entirely
+  const visibilityObserverOptions = {
+    root: fakeRevealContainer,
+    threshold: [0],
+  };
+  visibleSlideIntersectionObserver = new IntersectionObserver(
+    visibilityCallback,
+    visibilityObserverOptions
+  );
+  // Observe all actual sections, not the container sections of vertical stacks
+  for (const section of slideElementList) {
+    if (section.classList.contains("stack")) {
+      const subsections = section.querySelectorAll("section");
+      for (const subsection of subsections) {
+        visibleSlideIntersectionObserver.observe(subsection);
+      }
+    } else {
+      visibleSlideIntersectionObserver.observe(section);
+    }
+  }
+  fakeRevealContainer.addEventListener("scroll", determineMostVisibleSlide);
+}
+
+/**
+ * Create an intersection observer that observes all video, audio and iframe
+ * elements to only load their src if they are close to being visible.
+ * @returns Nothing
+ */
+function createSRCIntersectionObserver() {
+  if (srcIntersectionObserver) {
```

Comment:
bug: background videos/iframes are added dynamically each time handout mode is activated. if oberservers are not re-created these new elements are not observed (hence not loaded).